src/app/pages/CreateConnectorPage/StepNamespace.tsx

```diff
@@ -94,10 +107,58 @@ const ClustersGallery: FunctionComponent = () => {
     return t('evalNamespaceExpire', { hours, min });
   };
 
+  const prevResponse = usePrevious(response);
+
+  const getClusterName = (
+    clusterList: ConnectorClusterList,
+    clusterId: string
+  ) => {
+    return (
+      clusterList.items.find((cluster) => cluster.id === clusterId)?.name || ''
+    );
+  };
+
+  let getClusterInfo = useCallback(
+    (data) => {
+      const responseCopyItems = { ...response }.items!.map((item) => {
+        return {
+          ...item,
+          ...(!item.expiration && {
+            clusterName: getClusterName(data, item.cluster_id),
+          }),
+        };
+      });
+      setUpdateResponse({ ...response!, items: responseCopyItems });
+    },
+    [response]
+  );
+
+  const onClusterError = useCallback(() => {
+    const responseCopyItems = { ...response }.items!.map((item) => {
+      return {
+        ...item,
+        ...(!item.expiration && {
+          clusterName: t('clusterError'),
+        }),
+      };
+    });
+    setUpdateResponse({ ...response!, items: responseCopyItems });
+  }, [response]);
```

Comment:
The lint warning here is probably worth fixing.

Reply:
Done